src/components/Login/Login.test.tsx

```diff
@@ -0,0 +1,47 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { act } from "react-dom/test-utils";
+import { BrowserRouter } from "react-router-dom";
+import Login from "./Login";
+
+describe("Given a Login components", () => {
+  describe("When it's instantiated", () => {
+    test("Then it should show to inputs where the user can write", async () => {
+      const inputUserText = "Username";
+      const inputPasswordText = "Password";
+      const userWrite = "John Doe";
+      const passwordWrite = "123123";
+
+      render(
+        <BrowserRouter>
```

Comment:
You can extract this render method to a utils file, as you will use it across the whole App.

There you can config all the Providers you need and create different functions, for ex:

- reduxTestWrapperFactory(component, initialState)  --> This one will have redux provider
- formikTestWrapperFactory(component, initialFormikState, handleSumbit) --> Formik provider (so the components can use the formikContext hook

The function can be similar to this:

function whateverTestWrapperFactory(component) {
  return render(
    <BrowserRouter>
        <Provider1>
           <Provider2>
                {component}
           <Provider2>
        </Provider1>
    </BrowserRouter>
  )
}

Then you can invoke it like this:
const screen = whateverTestWrapperFactory(<MyComponent/>, {user:{logged: true}})

Reply:
Thanks for the idea. I will take it